### DISCUSSION TITLE
Testing

### DISCUSSION THREAD

**Discussion by tekumara** [acorn-io/runtime#575] [General]
Where do you see testing happening for an acorn app, specifically:

1. build tests, ie: things that need to run before the app container is built. Suppose these need to run in a container too. Would this happen outside acorn, or part of the acorn dev cycle? Either way, perhaps acorn's buildkit container could be used to build and run these tests? 

2. smoke tests post deployment: a job possibly? If so, would this be part of an acorn deployment? Is it possible to get feedback of their failure/success?

Thank you! 🙏

**Comment by ibuildthecloud** [comment]
Yes this is definitely a feature we have considered.  `acorn test [target] [args]` where target would correspond to a test job.  So it would look something like

```cue
containers: app: {
  build: "."
  if args.test {
    command: app "--enable-debug"
  }
}
tests: {
  quick: {
    build: {
        context: "."
        target: "test"
    }
    command: "make test"
    dependsOn: ["app"]
}
```

So the idea would be when you run `acorn test quick` it would build the container and you can specify a different multi-stage target, so maybe run your unit tests at this time.  And then the container will be started after all services you depend on are started and then acorn will wait until that job succeeds, failes, or times out.

Does this design seem good?
